app/src/main/java/org/lineageos/selfie/MainActivity.kt

```diff
@@ -537,74 +525,78 @@ class MainActivity : AppCompatActivity() {
+            // Start recording
+            recording = cameraController.startRecording(
+                outputOptions,
+                audioConfig,
+                cameraExecutor
+            ) {
+                val updateRecordingStatus = { enabled: Boolean, duration: Long ->
+                    // Hide mode buttons
+                    photoModeButton.isInvisible = enabled
+                    videoModeButton.isInvisible = enabled
+                    qrModeButton.isInvisible = enabled
+
+                    // Update duration text and visibility state
+                    videoDuration.text = TimeUtils.convertNanosToString(duration)
+                    videoDuration.isVisible = enabled
                 }
-                is VideoRecordEvent.Finalize -> {
-                    runOnUiThread {
-                        startShutterAnimation(ShutterAnimation.VideoEnd)
-                        updateRecordingStatus(false, 0)
+
+                when (it) {
+                    is VideoRecordEvent.Status -> runOnUiThread {
+                        updateRecordingStatus(true, it.recordingStats.recordedDurationNanos)
                     }
-                    cameraSoundsUtils.playStopVideoRecording()
-                    if (it.error != VideoRecordEvent.Finalize.ERROR_NO_VALID_DATA) {
-                        sharedPreferences.lastSavedUri = it.outputResults.outputUri
-                        Log.d(LOG_TAG, "Video capture succeeded: ${it.outputResults.outputUri}")
-                        tookSomething = true
+                    is VideoRecordEvent.Finalize -> {
+                        runOnUiThread {
+                            startShutterAnimation(ShutterAnimation.VideoEnd)
+                            updateRecordingStatus(false, 0)
+                        }
+                        cameraSoundsUtils.playStopVideoRecording()
+                        if (it.error != VideoRecordEvent.Finalize.ERROR_NO_VALID_DATA) {
+                            sharedPreferences.lastSavedUri = it.outputResults.outputUri
+                            Log.d(LOG_TAG, "Video capture succeeded: ${it.outputResults.outputUri}")
+                            tookSomething = true
+                        }
+                        recording = null
+                        takingSomething = false
                     }
-                    recording = null
                 }
             }
-        }
+        }, delayTime)
     }
 
     /**
      * Check if we can reinitialize the camera use cases
      */
     private fun canRestartCamera() = when (cameraMode) {
         // Disallow camera restart if we're taking a photo or if timer is running
-        CameraMode.PHOTO -> !isTakingPhoto && !countDownView.isVisible
+        CameraMode.PHOTO -> !takingSomething && !countDownView.isVisible
         // Disallow camera restart if a recording in progress or if timer is running
         CameraMode.VIDEO -> !cameraController.isRecording && !countDownView.isVisible
```

Comment:
This can now match photo mode conditions (`!takingSomething && !countDownView.isVisible`)